File: solutions/typescript/src/main6.ts
import * as fs from 'fs'

const [n, a, b] = fs.readFileSync('/dev/stdin', 'utf-8').split(' ').map(Number)

const check = (k: Number) => {
  const digitSum = k
    .toString()
    .split('')
    .map(Number)
    .reduce((sum, i) => sum + i)

  return a <= digitSum && digitSum <= b
}

let answer = 0
for (let i = 1; i <= n; i++) {
  if (check(i)) {
    answer += i
  }
}
console.log(answer)

File: solutions/rust/main6.rs
use proconio::input;

fn digit_sum(num: u32) -> u32 {
    if num == 0 {
        return 0;
    }

    return num % 10 + digit_sum(num / 10);
}

fn main() {
    input! {
        n: u32,
        a: u32,
        b: u32,
    }

    let answer = (1..=n).filter(|i| a <= digit_sum(*i) && digit_sum(*i) <= b).sum::<u32>();
    println!("{}", answer);
}
